### PR TITLE
[LDR] Check forwarded imports against the manifest before trying to load them

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-15729](https://jira.reactos.org/browse/CORE-15729)

### CODE REVIEW

File: dll/ntdll/ldr/ldrpe.c [hunk 1]
```diff
@@ -1133,13 +1133,46 @@ LdrpSnapThunk(IN PVOID ExportBase,
...
+                Status = RtlDosApplyFileIsolationRedirection_Ustr(TRUE,
+                                                                  &TempUString,
+                                                                  &LdrApiDefaultExtension,
+                                                                  &StaticString,
+                                                                  NULL,
+                                                                  &RedirectedImportName,
+                                                                  NULL,
+                                                                  NULL,
+                                                                  NULL);
+                /* Check success */
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrpe.c]: Seriously?

File: dll/ntdll/ldr/ldrpe.c [hunk 2]
```diff
@@ -1133,13 +1133,46 @@ LdrpSnapThunk(IN PVOID ExportBase,
...
+                /* Load the forwarder */
+                Status = LdrpLoadDll(Redirected,
                                      NULL,
                                      NULL,
-                                     &TempUString,
+                                     RedirectedImportName,
                                      &ForwarderHandle,
                                      FALSE);
+
+                /* free the temp string */
```
Comment (HBelusca) [inline dll/ntdll/ldr/ldrpe.c]: Capital F.

File: dll/ntdll/ldr/ldrpe.c (hunk 1 above)
Comment (learn-more) [reply dll/ntdll/ldr/ldrpe.c]: copypasta!

File: dll/ntdll/ldr/ldrpe.c (hunk 2 above)
Comment (learn-more) [reply dll/ntdll/ldr/ldrpe.c]: copypasta

Comment (yagoulas) [reply dll/ntdll/ldr/ldrpe.c]: Seriously?

Comment (yagoulas) [reply dll/ntdll/ldr/ldrpe.c]: @HBelusca if you reviewed the loader logic it would be more helpful

**Pull request merged**